src/AnchoredOverlay/AnchoredOverlay.tsx

```diff
@@ -0,0 +1,139 @@
+import React, {useCallback, useEffect, useMemo, useRef, useState} from 'react'
+import Overlay from '../Overlay'
+import Button, {ButtonProps} from '../Button'
+import {useFocusTrap} from '../hooks/useFocusTrap'
+import {useFocusZone} from '../hooks/useFocusZone'
+import {useAnchoredPosition, useRenderForcingRef} from '../hooks'
+import {uniqueId} from '../utils/uniqueId'
+
+export interface AnchoredOverlayProps {
+  /**
+   * A custom function component used to render the anchor element.
+   * Will receive the selected text as `children` prop when an item is activated.
+   * Uses a `DropdownButton` by default.
+   */
+  renderAnchor?: <T extends React.HTMLAttributes<HTMLElement>>(props: T) => JSX.Element
+
+  /**
+   * Children node to use as the contents of the overlay
+   */
+  children: React.ReactNode
```

Comment:
You shouldn't have to include `children` in props. See [this change](https://github.com/primer/components/pull/1181/files#diff-693c129d15d32a073c701ec06dc3d82242e49ff8f561501e2bd9f87d33892882) to ThemeProvider for an example.